tests/specs/stn/with-replay-error-mode.e2e.js

```diff
@@ -36,41 +34,40 @@ describe.withBrowsersMatching(notIE)('Trace error mode', () => {
   }
 
   it('switches to full mode when an error happens after page load', async () => {
-    await browser.url(await getReplayOnErrorUrl)
-    await browser.waitForAgentLoad()
-    await browser.waitUntil(() => browser.execute(async function () { // here for no-retry wdio stability
-      return await Object.values(newrelic.initializedAgents)[0].features.session_trace.onAggregateImported
-    }), { timeout: 5000 })
-    let startingMode = (await getTraceMode())[0]
-    expect(initSTReceived).toBeUndefined()
-    expect(startingMode).toEqual(MODE.ERROR)
-
-    await browser.execute(simulateErrorInBrowser)
-    await getFirstSTPayload
-    let modeAfterErr = (await getTraceMode())[0]
-    expect(modeAfterErr).toEqual(MODE.FULL)
+    await Promise.all([
+      browser.testHandle.expectResources(5001, true),
+      browser.url(await getReplayOnErrorUrl).then(() => browser.waitForFeatureAggregate('session_trace'))
+    ])
+    await expect(getTraceMode()).resolves.toEqual([MODE.ERROR, false])
+
+    const [resources] = await Promise.all([
+      browser.testHandle.expectResources(),
+      browser.execute(simulateErrorInBrowser)
+    ])
+    await expect(getTraceMode()).resolves.toEqual([MODE.FULL, false])
     // The loadEventEnd is part of PT and PNT resource entry and is a node created at start of page life.
-    expect(initSTReceived.request.body.res.find(node => node.n === 'loadEventEnd')).toEqual(expect.objectContaining({ n: 'loadEventEnd', o: 'document', t: 'timing' }))
+    expect(resources.request.body.res).toEqual(expect.arrayContaining([expect.objectContaining({
+      n: 'loadEventEnd', o: 'document', t: 'timing'
+    })]))
 
-    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy() // double check there's nothing wrong with full mdoe interval harvest
+    await expect(browser.testHandle.expectResources(5002)).resolves.toBeTruthy() // double check there's nothing wrong with full mode interval harvest
   })
 
   it('starts in full mode when an error happens before page load', async () => {
-    await browser.url(await getReplayOnErrorUrl)
-    await browser.execute(simulateErrorInBrowser)
+    let getFirstSTPayload = browser.testHandle.expectResources(5010)
```

Comment:
The 5002, 5010, 5011 feel oddly specific.  Is there a reason they are like this?

Reply:
Yeah, instead of making them all `5000`, it gets hard to tell which threw timeout when one of them fails. This is my way of making it easier to distinguish.

Reply:
The logging that happens as the test runs doesn't, so you'd have to wait until the whole test run finishes to sift and find out.